substrate/docs/deprecation-process.md

```diff
@@ -0,0 +1,75 @@
+# Deprecation Process
+
+This deprecation process makes sense while we don’t use [SemVer](https://semver.org/).
+After that, this process will most likely change.
+As deprecation and removal of existing code can happen on any release, we need to be mindful that external builders
+could be impacted by the changes we make.
+The deprecation process tries to mitigate this impact, while still keeping the developer experience, the DevEx, as
+smooth as possible.
+
+To start a deprecation process a new issue with the label `I11-deprecation` needs to be created for correct tracking.
+Then these are the actions to take:
+
+## Hard deprecate by adding a warning message
+
+The warning message shall include a removal month and year, which is suggested to be 6 months after the deprecation
+notice is released.
+This means that the code will be removed in a release within that month (or after, but never before). Please use this
+template, doing so makes it easy to search through the code base:
+
+```rust
+#[deprecated(note = "[DEPRECATED] will be removed after [DATE]. [ALTERNATIVE]")]
+```
+`[ALTERNATIVE]` won't always be possible but offer it if it is.
+
+E.g.
+```rust
+#[deprecated(note = "`GenesisConfig` will be removed after December 2023. Use `RuntimeGenesisConfig` instead.")]
+```
+
+Some pieces of code cannot be labeled as deprecated, like [reexports](https://github.com/rust-lang/rust/issues/30827)
+or [dispatchables](https://github.com/paritytech/polkadot-sdk/issues/182#issuecomment-1691684159), for instance.
+On cases like that we can only make a visible enough comment, and make sure that we [announce the deprecation properly](#announce-the-deprecation-and-removal).
+
+## Remove usage of the deprecated code in the code base
+
+Just make sure that we are not using the deprecated code ourselves.
+If you added the deprecation warning from the previous step, this can be done by making sure that warning is not shown
+when building the code.
+
+## Update examples and tutorials
+
+Make sure that the rust docs are updated.
+We also want [https://docs.substrate.io/](https://docs.substrate.io/) to be updated, you can open an issue on [https://github.com/substrate-developer-hub/substrate-docs](https://github.com/substrate-developer-hub/substrate-docs).
```

Comment:
In the short term, I'd say that it is primarily the responsibility of the author of the PR here to update these docs. If not feasible, they can reside to opening an issue.